### ISSUE TITLE
Skipping fields when parsing large classes

### ISSUE THREAD

**Issue by Skeptick** [Kotlin/kotlinx.serialization#164]
When I try to parse a JSON string, one of the values in the target class is at 33 positions or further, this value is always `null`

> При попытке распарсить JSON строку, одно из значений которой в целевом классе находится на 33 позиции или дальше, это значение всегда `null`. Более понятно на примере ниже.

```kotlin

@Serializable
data class LargeClass(
    @Optional @SerialName("field0") val field0: String? = null,
    @Optional @SerialName("field1") val field1: String? = null,
    @Optional @SerialName("field2") val field2: String? = null,
    @Optional @SerialName("field3") val field3: String? = null,
    @Optional @SerialName("field4") val field4: String? = null,
    @Optional @SerialName("field5") val field5: String? = null,
    @Optional @SerialName("field6") val field6: String? = null,
    @Optional @SerialName("field7") val field7: String? = null,
    @Optional @SerialName("field8") val field8: String? = null,
    @Optional @SerialName("field9") val field9: String? = null,
    @Optional @SerialName("field10") val field10: String? = null,
    @Optional @SerialName("field11") val field11: String? = null,
    @Optional @SerialName("field12") val field12: String? = null,
    @Optional @SerialName("field13") val field13: String? = null,
    @Optional @SerialName("field14") val field14: String? = null,
    @Optional @SerialName("field15") val field15: String? = null,
    @Optional @SerialName("field16") val field16: String? = null,
    @Optional @SerialName("field17") val field17: String? = null,
    @Optional @SerialName("field18") val field18: String? = null,
    @Optional @SerialName("field19") val field19: String? = null,
    @Optional @SerialName("field20") val field20: String? = null,
    @Optional @SerialName("field21") val field21: String? = null,
    @Optional @SerialName("field22") val field22: String? = null,
    @Optional @SerialName("field23") val field23: String? = null,
    @Optional @SerialName("field24") val field24: String? = null,
    @Optional @SerialName("field25") val field25: String? = null,
    @Optional @SerialName("field26") val field26: String? = null,
    @Optional @SerialName("field27") val field27: String? = null,
    @Optional @SerialName("field28") val field28: String? = null,
    @Optional @SerialName("field29") val field29: String? = null,
    @Optional @SerialName("field30") val field30: String? = null,
    @Optional @SerialName("field31") val field31: String? = null,
    @Optional @SerialName("field32") val field32: String? = null,
    @Optional @SerialName("field33") val field33: String? = null,
    @Optional @SerialName("field34") val field34: String? = null,
    @Optional @SerialName("field35") val field35: String? = null,
    @Optional @SerialName("field36") val field36: String? = null,
    @Optional @SerialName("field37") val field37: String? = null,
    @Optional @SerialName("field38") val field38: String? = null,
    @Optional @SerialName("field39") val field39: String? = null,
    @Optional @SerialName("field40") val field40: String? = null
)

val jsonString = """{"field31":"value31","field32":"value32"}"""
JSON.parse<LargeClass>(jsonString) // > LargeClass([...], field31=value31, field32=null, [...])
```

**Comment by pdvrieze** [comment]
As the system works now the writing will always write all fields. Optional is only for reading. It is possible to write a custom serializer with custom annotation that will actually somehow try to not write fields with default values, but it needs a way to determine what is the default. That is hard to do generically.

**Comment by InsanusMokrassar** [comment]
Is there any update for this issue?

**Comment by sandwwraith** [comment]
@InsanusMokrassar this should be fixed in 1.3.30

**Comment by InsanusMokrassar** [comment]
Yes, it is fixed, thank you, @sandwwraith
